app/controllers/application_controller.rb

```diff
@@ -147,4 +147,13 @@ def respond_with_error(code)
       format.json { render json: { error: Rack::Utils::HTTP_STATUS_CODES[code] }, status: code }
     end
   end
+
+  def handle_limit_as_alert(exc)
+    # We want to handle rate limit errors similar to how they are handled
+    # when using the API, called by the frontend, i.e. with alerts instead of 
+    # rendering an error page
+    flash[:alert] = 'Rate limit: you have done this action too many times! Try
+                     again on ' + (Time.now + exc.cooldown.seconds).in_time_zone('Berlin').to_s + ' .'
+    redirect_back fallback_location: root_path
+  end
```

Comment:
Three thoughts:

1. Would be nice if this string could be translatable. 
2. I also think it is a bit too confrontational for the invite use case, would be nice if we could have a message more like "Sorry you can only create 1 invite per week. Try again after [date]"
3. Also, is it possible to use the already configured timezone here instead of hard-coding Berlin?

Reply:
Completely agree with all point, the message was a placeholder really (all HTTP 429 were handled by that, not only the ones generated by invites). I'll specialize it a little and look into the time zone and translation things as well.